utils/gui/Searchbar.js

```diff
@@ -0,0 +1,252 @@
+        input.setAttribute('placeholder', 'Search location');
+        input.style.height = `${options.height || options.size || DEFAULT_OPTIONS.height}px`;
+        input.style.fontSize = `${options.fontSize || DEFAULT_OPTIONS.fontSize}px`;
+        form.appendChild(input);
+
+        // currentFocus variable stores the index of the suggestions that is focused by user, either with mouse or arrow
+        // keys.
+        let currentFocus;
+
+
+
+        // ----------  BUILD AUTOCOMPLETION SUGGESTIONS LIST WHEN TYPING THE SEARCHBAR INPUT : ----------
+
+        input.addEventListener('input', () => {
+            const value = input.value;
+
+            // Close any already opened list of autocompleted values
+            this.eraseSuggestionList(form);
+
+            currentFocus = -1;
+
+            if (!value) { return false; }
+
+            geocodingOptions.url.searchParams.set('text', value);
+
+            Fetcher.json(geocodingOptions.url).then((geocodingResult) => {
+                const resultArray = geocodingOptions.parser(geocodingResult);
+
+                for (
+                    let i = 0;
+                    i < Math.min(
+                        resultArray.length, options.maxSuggestionNumber || DEFAULT_OPTIONS.maxSuggestionNumber,
+                    );
+                    i++
+                ) {
+                    const location = resultArray[i];
+
+                    const index = location.toUpperCase().indexOf(value.toUpperCase());
+
+                    if (index > -1) {
+                        const autocompleteItem = document.createElement('div');
+
+                        autocompleteItem.style.minHeight = input.style.height;
+                        autocompleteItem.style.fontSize = `${options.fontSize || DEFAULT_OPTIONS.fontSize}px`;
+
+                        // Make the matching letters bold.
+                        const start = location.slice(0, index);
+                        const bold = location.slice(index, index + value.length);
+                        const end = location.slice(index + value.length, location.length);
+
+                        autocompleteItem.innerHTML = `<p>${start}<strong>${bold}</strong>${end}</p>`;
+                        // Store the current location value as an attribute of autocompleteItem div.
+                        autocompleteItem.setAttribute('location', location);
+
+                        form.appendChild(autocompleteItem);
+
+                        // eslint-disable-next-line no-loop-func
+                        autocompleteItem.addEventListener('mouseover', () => {
+                            removeAllActives(form.children);
+                            currentFocus = i;
+                            autocompleteItem.classList.add('active');
+                        });
+
+                        autocompleteItem.addEventListener('click', () => {
+                            geocodingOptions.whenSelected(geocodingResult.results[i]);
```

Comment:
could you use the map instead of `geocodingResult`

Reply:
Done